app/app.cpp

```diff
@@ -1,7 +1,102 @@
 #include "app.h"
 #include <algorithm>
+#include <cctype>
 
-namespace vsite::oop::v3
-{
+namespace vsite::oop::v3 {
 
-}
\ No newline at end of file
+    student::student(const std::string& name, int grade)
+        : name(name), grade(grade)
+    {
+    }
+
+    results::results(size_t capacity)
+        : capacity(capacity), count(0)
+    {
+        students = new student[capacity];
+    }
+
+    results::~results()
+    {
+        delete[] students;
+    }
+
+    void results::add(const student& s)
+    {
+        if (count < capacity)
+            students[count++] = s;
+    }
+
+    unsigned int results::has_grade(int g) const
+    {
+        unsigned int total = 0;
+        for (int i = 0; i < count; ++i)
+            if (students[i].grade == g)
+                ++total;
+        return total;
+    }
+
+    unsigned int results::starts_with_letter(char l) const
+    {
+        unsigned int total = 0;
+        for (int i = 0; i < count; ++i)
+        {
+            if (!students[i].name.empty())
+            {
+                char first = std::tolower(students[i].name[0]);
+                if (first == std::tolower(l))
```

Comment:
Zašto bi se `tolower` za `l` radio svaki puta u petlji, zašto ne jednom prije petlje?